lib/kochiku/worker.rb

```diff
@@ -60,6 +60,32 @@ def msg2str(msg)
       def build_strategy
         @build_strategy ||= BuildStrategyFactory.get_strategy(settings.build_strategy)
       end
+
+      def instance_type
+        @instance_type ||= begin
+          path = File.join(ENV["HOME"], 'instance_type.txt')
+          if File.file?(path)
+            instance_type = File.read(path)
+          elsif settings.in_ec2
+            instance_type = request_instance_type_from_ec2
+            File.write(path, instance_type) if instance_type
+          end
+          instance_type
+        end
+      end
+
+      def request_instance_type_from_ec2
+        begin
+          instance_type = RestClient::Request.execute(method: :get,
+                                                      url: "http://169.254.169.254/latest/meta-data/instance-type",
+                                                      timeout: 15,
+                                                      open_timeout: 15).body
```

Comment:
15 seconds is excessively long. 5 seconds sounded good. Additionally we only need to specify `open_timeout`. No way will reading this little bit of text take so long that we need a custom `timeout`.